### PR TITLE
Powerloader refactor

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

Refactors powerloaders to be something more akin of the `attack_x` procs so we can avoid snowflaking the power clamp `after_attack` for each and every type of object it's supposed to be able to pick up.

Made being pickup-able by a powerloader a flag.
## Why It's Good For The Game

I don't like old powerloader code which everyone has just been adding more snowflake onto instead of fixing. Fixes that.
## Changelog
:cl:
refactor: Refactored the powerloader
/:cl:

### CODE REVIEW

**Comment by zzurge** [comment]
Power loader enjoyers finally eating something

File: code/_globalvars/bitfields.dm [hunk 1]
```diff
@@ -566,7 +566,9 @@ GLOBAL_LIST_INIT(bitfields, list(
 		"INTERACT_REQUIRES_DEXTERITY" = INTERACT_REQUIRES_DEXTERITY,
 		"INTERACT_CHECK_INCAPACITATED" = INTERACT_CHECK_INCAPACITATED,
 		"INTERACT_UI_INTERACT" = INTERACT_UI_INTERACT,
-		"INTERACT_SILICON_ALLOWED" = INTERACT_SILICON_ALLOWED
+		"INTERACT_SILICON_ALLOWED" = INTERACT_SILICON_ALLOWED,
+		"INTERACT_POWERLOADER_PICKUP_ALLOWED" = INTERACT_POWERLOADER_PICKUP_ALLOWED,
+		"INTERACT_POWERLOADER_PICKUP_ALLOWED_BYPASS_ANCHOR" = INTERACT_POWERLOADER_PICKUP_ALLOWED_BYPASS_ANCHOR,
```
Comment (TiviPlus) [inline code/_globalvars/bitfields.dm:571]: These being interact flags feels very odd considering they dont interact with any of the actual interact flags, whyd you do this?

**Comment by TiviPlus** [comment]
rest lgtm

File: code/_globalvars/bitfields.dm (hunk 1 above)
Comment (ivanmixo) [reply code/_globalvars/bitfields.dm:571]: Wasn't sure of a better better flag var to put them in

**Pull request merged**